src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs

```diff
@@ -10,100 +9,261 @@
+            var viewSymbols = Views.Select(v => v.Symbol);
 
-            List<GraphInitializerInfo> initializers = new List<GraphInitializerInfo>();
+			if (Type == GraphType.PXGraph)
+			{
+				var graphDelegates = Symbol.GetViewDelegatesFromGraph(viewSymbols, _pxContext, _cancellation);
 
-            if (Type == GraphType.PXGraph)
-            {
-                IEnumerable<GraphInitializerInfo> ctrs = Symbol.GetDeclaredInstanceConstructors(_cancellation)
-                                                         .Select(ctr => new GraphInitializerInfo(GraphInitializerType.InstanceCtr, ctr.Node, ctr.Symbol));
-                initializers.AddRange(ctrs);
+                foreach (var del in graphDelegates)
+                {
+                    viewDelegateByNameDictionary[del.Symbol.Name] = new DataViewDelegateInfo(del.Node, del.Symbol);
+                }
             }
-            else if (Type == GraphType.PXGraphExtension)
+            else
             {
-                (MethodDeclarationSyntax node, IMethodSymbol symbol) = Symbol.GetGraphExtensionInitialization(_pxContext, _cancellation);
+                var extDelegates = Symbol.GetViewDelegatesFromGraphExtensionAndBaseGraph(viewSymbols, _pxContext, _cancellation);
 
-                if (node != null && symbol != null)
+                foreach (var del in extDelegates)
                 {
-                    initializers.Add(new GraphInitializerInfo(GraphInitializerType.InitializeMethod, node, symbol));
+                    var delegateInfo = del.Base == null
+                        ? new DataViewDelegateInfo(del.Item.Node, del.Item.Symbol)
+                        : new DataViewDelegateInfo(del.Item.Node, del.Item.Symbol, new DataViewDelegateInfo(del.Base.Item.Node, del.Base.Item.Symbol));
+
+                    viewDelegateByNameDictionary[del.Item.Symbol.Name] = delegateInfo;
                 }
             }
 
-            Initializers = initializers.ToImmutableArray();
+            return viewDelegateByNameDictionary.ToImmutableDictionary();
         }
 
+		private ImmutableDictionary<string, ActionInfo> GetActions()
+		{
+			if (Type == GraphType.None)
+				return ImmutableDictionary.Create<string, ActionInfo>(StringComparer.OrdinalIgnoreCase);
+
+			var systemActionsRegister = new PXSystemActions.PXSystemActionsRegister(_pxContext);
+			var rawActionInfos = Type == GraphType.PXGraph
+				? Symbol.GetActionSymbolsWithTypesFromGraph(_pxContext)
+				: Symbol.GetActionsFromGraphExtensionAndBaseGraph(_pxContext);
+
+			return rawActionInfos.ToLookup(a => a.Item.ActionSymbol.Name, StringComparer.OrdinalIgnoreCase)
+								 .ToImmutableDictionary(group => group.Key,
+														group => CreateActionInfo(group.First()),
+														keyComparer: StringComparer.OrdinalIgnoreCase);
+
+			//--------------------------------------------------------Local Function--------------------------------------------
```

Comment:
I think this comment looks ugly. What do you think, can we just keep local functions in the end of methods without any comment?

Reply:
ok